sonic_platform_base/sonic_xcvr/mem_maps/public/cmisTargetFWUpgrade.py

```diff
@@ -0,0 +1,24 @@
+"""
+    cmisTargetFWUpgrade.py
+
+    Implementation of memory map for CMIS based modules supporting firmware
+    upgrade of remote target from the local target itself.
+"""
+
+from .cmis import CmisMemMap
+from ...fields.xcvr_field import (
+    NumberRegField,
+    RegGroupField,
+)
+from ...fields import consts
+
+class CmisTargetFWUpgradeMemMap(CmisMemMap):
+    def __init__(self, codes):
+        super().__init__(codes)
+
+        self.VENDOR_CUSTOM = RegGroupField(consts.VENDOR_CUSTOM,
```

Comment:
@mihirpat1 can we keep this in Credo specific memmap?

Reply:
@prgeor I have addressed this now.